File: scala-spark/src/main/scala/com/myCompany/ks2/Kata2_2.scala
package com.myCompany.ks2

import com.myCompany.Launch
import org.apache.spark.rdd.RDD
import org.apache.spark.sql.SparkSession

import java.nio.file.Paths

object Kata2_2 {
  def main(args: Array[String]): Unit = {
    val spark: SparkSession = SparkSession.builder.appName("scala-Basics").master("local[*]")
      .getOrCreate

    val pathToDataFile: String = Paths.get(Launch.path_to_directory, "SalesJan2009.csv").toString
    val lines: RDD[String] = spark.read.textFile(pathToDataFile).rdd

    lines.take(10).foreach((e: String) => System.out.println(e))

    val linesNoHeader: RDD[String] = lines.filter((e: String) => !(e.startsWith("Transaction_date")))

    println(s"initial number of lines, ${lines.count}")

    val csv_sep: String = ";"
    val price_index: Int = 2
    val cleanedData: RDD[String] = linesNoHeader.filter((e: String) => e.split(csv_sep)(price_index).matches("^\\d+$"))
    println(s"final number of lines, ${cleanedData.count}")

    spark.stop()
  }
}

File: java-basics/src/main/java/com/myCompany/ks2/Kata2_2.java
package com.myCompany.ks2;

import com.myCompany.Launch;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import java.io.IOException;
import java.nio.file.Paths;

public class Kata2_2 {
    public static void main(String[] args) throws IOException {
        // creating java spark context
        SparkConf sparkConf = new SparkConf().setAppName("Java-Advanced").setMaster("local[*]");
        JavaSparkContext jsc = new JavaSparkContext(sparkConf);

        String pathToDataFile = Paths.get(Launch.path_to_directory, "SalesJan2009.csv").toString();
        JavaRDD<String> lines = jsc.textFile(pathToDataFile);

        lines.take(10).forEach(e-> System.out.println(e));

        JavaRDD<String> linesNoHeader = lines.filter(e -> !e.startsWith("Transaction_date"));

        System.out.println(String.format("initial number of lines %d", lines.count()));

        String csv_sep = ";";
        int price_index = 2;
        JavaRDD<String> cleanedData = linesNoHeader
                .filter(e->e.split(csv_sep)[price_index].matches("^\\d+$"));

        System.out.println(String.format("final number of lines %d", cleanedData.count()));

        jsc.stop();
    }
}
